nifi-nar-bundles/nifi-kite-bundle/nifi-kite-processors/src/main/java/org/apache/nifi/processors/kite/InferAvroSchema.java

```diff
@@ -0,0 +1,454 @@
+            .name("CSV escape string")
+            .description("This property only applies to CSV content type. String that represents an escape sequence" +
+                    " in the CSV FlowFile content data.")
+            .required(true)
+            .defaultValue("\\")
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor QUOTE_STRING = new PropertyDescriptor.Builder()
+            .name("CSV quote string")
+            .description("This property only applies to CSV content type. String that represents a literal quote" +
+                    " character in the CSV FlowFile content data.")
+            .required(true)
+            .defaultValue("'")
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RECORD_NAME = new PropertyDescriptor.Builder()
+            .name("Avro Record Name")
+            .description("Value to be placed in the Avro record schema \"name\" field.")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor CHARSET = new PropertyDescriptor.Builder()
+            .name("Charset")
+            .description("Character encoding of CSV data.")
+            .required(true)
+            .defaultValue("UTF-8")
+            .addValidator(StandardValidators.CHARACTER_SET_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor PRETTY_AVRO_OUTPUT = new PropertyDescriptor.Builder()
+            .name("Pretty Avro Output")
+            .description("If true the Avro output will be formatted.")
+            .required(true)
+            .defaultValue("true")
+            .allowableValues("true", "false")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor NUM_RECORDS_TO_ANALYZE = new PropertyDescriptor.Builder()
+            .name("Number of records to analyze")
+            .description("This property only applies to JSON content type. The number of JSON records that should be" +
+                    " examined to determine the Avro schema. The higher the value the better chance kite has of detecting" +
+                    " the appropriate type. However the default value of 10 is almost always enough.")
+            .required(true)
+            .defaultValue("10")
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder().name("success")
+            .description("Successfully created Avro schema for CSV data.").build();
```

Comment:
Should this read for "CSV/JSON data" or something of that nature? See "CSV data" mentioned for all of the relationships except for "unsupported content"